core/src/main/java/tc/oc/pgm/spawner/SpawnerModule.java

```diff
@@ -49,58 +54,82 @@ public SpawnerModule parse(MapFactory factory, Logger logger, Document doc)
         Duration minDelay = XMLUtils.parseDuration(minDelayAttr, delay);
         Duration maxDelay = XMLUtils.parseDuration(maxDelayAttr, delay);
 
         if (maxDelay.compareTo(minDelay) <= 0 && minDelayAttr != null && maxDelayAttr != null) {
-          throw new InvalidXMLException("Max-delay must be longer than min-delay", element);
+          throw new InvalidXMLException("Max-delay must be longer than min-delay", spawnerEl);
         }
 
         int maxEntities =
             XMLUtils.parseNumber(
-                element.getAttribute("max-entities"), Integer.class, Integer.MAX_VALUE);
+                spawnerEl.getAttribute("max-entities"), Integer.class, Integer.MAX_VALUE);
         Filter playerFilter =
-            filterParser.parseFilterProperty(element, "filter", StaticFilter.ALLOW);
+            filterParser.parseFilterProperty(spawnerEl, "filter", StaticFilter.ALLOW);
 
         List<Spawnable> objects = new ArrayList<>();
-        for (Element spawnable :
-            XMLUtils.getChildren(
-                element, "item")) { // TODO Add more types of spawnables once entity parser is built
-          ItemStack stack = kitParser.parseItem(spawnable, false);
-          SpawnableItem item = new SpawnableItem(stack, numericID);
+        for (Element itemEl : XMLUtils.getChildren(spawnerEl, "item")) {
+          ItemStack stack = kitParser.parseItem(itemEl, false);
+          SpawnableItem item = new SpawnableItem(stack, "spawner-" + numericId.get());
           objects.add(item);
         }
 
+        ImmutableList.Builder<PotionEffect> chBuilder = ImmutableList.builder();
+        for (Element potionEl : XMLUtils.getChildren(spawnerEl, "potion")) {
+          for (Element potionChild : potionEl.getChildren("effect")) {
+            chBuilder.add(XMLUtils.parsePotionEffect(new InheritingElement(potionChild)));
+          }
+          ImmutableList<PotionEffect> potionChildren = chBuilder.build();
+          if (potionChildren.isEmpty()) {
+            throw new InvalidXMLException("Expected child effects, but found none", spawnerEl);
+          }
+          int damageValue = 0;
+          if (potionEl.getAttribute("damage") != null) {
+            damageValue = XMLUtils.parseNumber(potionEl.getAttribute("damage"), Integer.class, 0);
+          } else {
+            for (PotionEffect potionEffect : potionChildren) {
+              // PotionType lists "true" potions, PotionEffectType "potionEffect.getType()" lists
+              // all possible status effects (ie wither, blindness, etc)
+              // Use the first listed PotionType for potion color
+              if (PotionType.getByEffect(potionEffect.getType()) != null) {
+                damageValue = PotionType.getByEffect(potionEffect.getType()).getDamageValue();
+                break;
+              }
+            }
+          }
+          objects.add(
+              new SpawnablePotion(potionChildren, damageValue, "spawner-" + numericId.get()));
```

Comment:
```suggestion
              new SpawnablePotion(potionChildren, damageValue, id));
```
A string id should be created beforehand, taken from the XML "id" attribute if one is available, only if null do you put an autogenerated id